streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/impl/VariableServiceImpl.java

```diff
@@ -0,0 +1,120 @@
+import org.apache.streampark.console.system.mapper.VariableMapper;
+import org.apache.streampark.console.system.service.VariableService;
+
+import com.baomidou.mybatisplus.core.conditions.query.LambdaQueryWrapper;
+import com.baomidou.mybatisplus.core.metadata.IPage;
+import com.baomidou.mybatisplus.extension.plugins.pagination.Page;
+import com.baomidou.mybatisplus.extension.service.impl.ServiceImpl;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Slf4j
+@Service
+@Transactional(propagation = Propagation.SUPPORTS, readOnly = true, rollbackFor = Exception.class)
+public class VariableServiceImpl extends ServiceImpl<VariableMapper, Variable> implements VariableService {
+
+    @Autowired
+    private ApplicationService applicationService;
+
+    @Autowired
+    private CommonService commonService;
+
+    @Override
+    @Transactional(rollbackFor = Exception.class)
+    public void createVariable(Variable variable) throws Exception {
+        if (isExists(variable)) {
+            throw new ApiAlertException("Sorry, the variable code already exists.");
+        }
+        variable.setCreator(commonService.getCurrentUser().getUserId());
+        this.save(variable);
+    }
+
+    @Override
+    public IPage<Variable> findVariables(Variable variable, RestRequest request) {
+        Page<Variable> page = new Page<>();
+        page.setCurrent(request.getPageNum());
+        page.setSize(request.getPageSize());
+        return this.baseMapper.findVariable(page, variable);
+    }
+
+    @Override
+    public void updateVariable(Variable variable) throws Exception {
+        Variable findVariable = this.findByVariableCode(variable.getTeamId(), variable.getVariableCode());
+        if (findVariable == null) {
+            throw new ApiAlertException("Sorry, the variable code already exists.");
+        }
+        if (findVariable.getId().longValue() != variable.getId().longValue()) {
+            throw new ApiAlertException("Sorry, the variable id is inconsistent.");
+        }
+        updateById(variable);
+    }
+
+    @Override
+    public void deleteVariable(Variable variable) throws Exception {
+        Variable findVariable = this.findByVariableCode(variable.getTeamId(), variable.getVariableCode());
+        if (findVariable == null) {
+            throw new ApiAlertException("Sorry, the variable code already exists.");
+        }
+        if (findVariable.getId().longValue() != variable.getId().longValue()) {
+            throw new ApiAlertException("Sorry, the variable id is inconsistent.");
+        }
+        List<Application> dependApplications = this.findDependByCode(variable);
+        if (!(dependApplications == null || dependApplications.isEmpty())) {
```

Comment:
How about `dependApplications != null && !dependApplications.isEmpty()`?

Or could we ensure the return value of `findDependByCode(variable)` is always not null? We can return an empty list~. 

If we can, the condition will be more simple, it can be `!dependApplications.isEmpty()`

Reply:
I deleted this judgment, I will judge in the next PR, because it is indeed related to the next PR.